meilisearch-http/src/routes/indexes/documents.rs

```diff
@@ -227,26 +227,51 @@ async fn document_addition(
 
     let (uuid, mut update_file) = index_scheduler.create_update_file()?;
 
-    // TODO: this can be slow, maybe we should spawn a thread? But the payload isn't Send+Sync :weary:
-    // push the entire stream into a `Vec`.
-    // If someone sends us a never ending stream we're going to block the thread.
-    // TODO: Maybe we should write it to a file to reduce the RAM consumption
-    // and then reread it to convert it to obkv?
-    let mut buffer = Vec::new();
+    let err: Result<SummarizedTaskView, MeilisearchHttpError> =
+        Err(MeilisearchHttpError::Payload(ReceivePayloadErr));
+
+    let temp_file = match NamedTempFile::new() {
+        Ok(temp_file) => temp_file,
+        Err(e) => {
+            error!("create a temporary file error: {}", e);
+            return err;
+        }
+    };
+    debug!("temp file path: {:?}", temp_file.as_ref());
+    let buffer_file = match temp_file.reopen() {
+        Ok(buffer_file) => buffer_file,
+        Err(e) => {
+            error!("reopen payload temporary file error: {}", e);
+            return err;
+        }
+    };
+    let mut buffer = BufWriter::new(buffer_file);
+    let mut buffer_write_size: usize = 0;
     while let Some(bytes) = body.next().await {
-        buffer.extend_from_slice(&bytes?);
+        match buffer.write(&bytes?) {
+            Ok(size) => buffer_write_size += size,
+            Err(e) => {
+                error!("bufWriter write error: {}", e);
+                return err;
+            }
+        }
     }
-    if buffer.is_empty() {
+
+    if let Err(e) = buffer.flush() {
+        error!("bufWriter flush error: {}", e);
+        return err;
+    };
```

Comment:
We're in an async context here, creating a file is already far from perfect, but doing write and flush is absolutely forbidden!

You should try to use the tokio io-util API for all these operations;
- [BufWriter](https://docs.rs/tokio/latest/tokio/io/struct.BufWriter.html)
- [flush](https://docs.rs/tokio/latest/tokio/io/trait.AsyncWriteExt.html#method.flush)

Reply:
ok. I will try it with the tokio io-util API